src/core/filesystem.cc

```diff
@@ -1470,7 +1470,8 @@ S3FileSystem::FileModificationTime(const std::string& path, int64_t* mtime_ns)
   // If request succeeds, copy over the modification time
   auto head_object_outcome = client_.HeadObject(head_request);
   if (head_object_outcome.IsSuccess()) {
-    *mtime_ns = head_object_outcome.GetResult().GetLastModified().Millis();
+    *mtime_ns =
+        head_object_outcome.GetResult().GetLastModified().Millis() * 1000000;
```

Comment:
Use NANOS_PER_MILLIS from constants.h. Or better, is there a Nanos() call that can replace Millis()?

Reply:
Will use NANOS_PER_MILLIS. No there is no Nanos() call for the AWS DateTime object
https://sdk.amazonaws.com/cpp/api/0.12.9/df/dc4/class_aws_1_1_utils_1_1_date_time.html